source/Date.cpp

```diff
@@ -49,23 +52,57 @@ Date::Date(int day, int month, int year)
 // Convert a date to a string.
 const string &Date::ToString() const
 {
+	string dateFormat = Preferences::DateFormat();
+
+	if(dateFormat != dateFormatInUse)
+	{
+		dateFormatInUse = dateFormat;
+		str = "";
+	}
+
 	// Because this is a somewhat "costly" operation, cache the result. The
 	// cached value is discarded if the date is changed.
-	if(date && str.empty())
+	if((date && str.empty()))
```

Comment:
Why are you dding brackets here?
```suggestion
	if(date && str.empty())
```

Reply:
I was refactoring that statement. Until I realized my refactor was terrible and changed it back. Oops, forgot the brackets.